app/models/manageiq/providers/openstack/cloud_manager/upload_image_workflow.rb

```diff
@@ -0,0 +1,76 @@
+class ManageIQ::Providers::Openstack::CloudManager::UploadImageWorkflow < Job
+  def self.create_job(ext_management_system_id, image_id, url)
+    options = {
+      :ext_management_system_id => ext_management_system_id,
+      :url                      => url,
+      :image_id                 => image_id,
+    }
+
+    super(name, options)
+  end
+
+  def pre_upload_image
+    queue_signal(:upload_image)
```

Comment:
Unless you intend to override this in a child class this doesn't look necessary, you can just have your `:start` signal `=> :upload_image`